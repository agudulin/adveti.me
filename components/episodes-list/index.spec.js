import React from 'react'
import renderer from 'react-test-renderer'

import EpisodesList from './'

jest.mock('react-dom')

const basicProps = {
  season: 1,
  episodes: [{
    id: '01a',
    poster: 'poster-url-1',
    name: 'episode-name-1',
    videos: [{
      url: 'video-url-1',
      name: 'video-name-1'
    }]
  }, {
    id: '01b',
    poster: 'poster-url-2',
    name: 'episode-name-2',
    videos: [{
      url: 'video-url-1',
      name: 'video-name-1'
    }]
  }]
}

describe('EpisodesList', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <EpisodesList {...basicProps} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
