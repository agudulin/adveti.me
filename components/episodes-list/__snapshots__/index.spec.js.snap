exports[`EpisodesList renders correctly 1`] = `
<ul
  data-jsx={3532644266}>
  <li
    data-jsx={3532644266}>
    <img
      className="poster"
      data-jsx={3532644266}
      src="https://api.adveti.me/season/1/episode/01a/poster.jpg" />
    <h3
      data-jsx={3532644266}>
      01a - episode-name-1
    </h3>
    <div
      className="video-links"
      data-jsx={3532644266}>
      <a
        className="video-links__item"
        data-jsx={3532644266}
        href="video-url-1">
        video-name-1
      </a>
    </div>
  </li>
  <li
    data-jsx={3532644266}>
    <img
      className="poster"
      data-jsx={3532644266}
      src="https://api.adveti.me/season/1/episode/01b/poster.jpg" />
    <h3
      data-jsx={3532644266}>
      01b - episode-name-2
    </h3>
    <div
      className="video-links"
      data-jsx={3532644266}>
      <a
        className="video-links__item"
        data-jsx={3532644266}
        href="video-url-1">
        video-name-1
      </a>
    </div>
  </li>
</ul>
`;
